test/blackbox/common/DDSBlackboxTestsDataReader.cpp

```diff
@@ -414,6 +414,80 @@ TEST(DDSDataReader, default_qos_large_history_depth)
     ASSERT_TRUE(reader.isInitialized());
 }
 
+/*
+* This test creates a DataReader using DATAREADER_QOS_USE_TOPIC_QOS and checks if DataReader QoS
+* are correctly set from Topic QoS. It also checks if one of the QoS that is not in common with Topic QoS
+* is correctly set as the default one.
+*/
+TEST(DDSDataReader, datareader_qos_use_topic_qos)
```

Comment:
In case of a creation of a datareader with `DATAREADER_QOS_USE_TOPIC_QOS` the qos that are not in common with the topic qos are set as the default ones